### ISSUE TITLE
OPDS catalog fails with 500 on OC9

### ISSUE THREAD

**Issue by ZafiRUS** [Yetangitu/owncloud-apps#13]
OC9. Browsing into files_opds results in 500 error.
Owncloud log reads the following:
```
Error	index	OCP\Files\NotFoundException: File with id "0" has not been found.
/var/www/cloud/htdocs/lib/private/files/filesystem.php - line 903: OC\Files\View->getPath('root')
/var/www/cloud/htdocs/apps/files_opds/index.php - line 48: OC\Files\Filesystem getPath('root')
/var/www/cloud/htdocs/lib/private/route/route.php - line 154): runtime-created function(1) require_once('/var/www/cloud/...')
[internal function] __lambda_func(Array)
/var/www/cloud/htdocs/lib/private/route/router.php - line 273: call_user_func('\x00lambda_543', Array)
/var/www/cloud/htdocs/lib/base.php - line 873: OC\Route\Router->match('/apps/files_opd...')
/var/www/cloud/htdocs/index.php - line 39: OC handleRequest()
{main}
```

Server gentoo-4.4.1, nginx-1.8.1, php-fpm-5.6.18

**Comment by polkillas** [comment]
Same error here (Raspbian Jessie, Apacha 2.4.10, Owncloud 9.0 fresh install, php 5.6.17).

**Comment by Yetangitu** [comment]
I'll look into this ASAP, it works_for_me (tm) (on Debian Sid, OC9, PHP5 and PHP7, PostgreSQL) but this is the second (or third, depending on whether one of you goes by the name of 'elvenbone' on the OC forums) report I get on this issue.

**Comment by rdp5008** [comment]
I am also getting a similar error when trying to access files_opds.

<pre><code>Error 	index Exception: 
{"Exception":"OCP\\Files\\NotFoundException","Message":"File with id \"0\" has not been found.",
"Code":0,"Trace":"#0 \/var\/www\/owncloud\/lib\/private\/files\/filesystem.php(903): OC\\Files\\View->getPath('root')\
n#1 \/var\/www\/owncloud\/apps\/files_opds\/index.php(48): OC\\Files\\Filesystem::getPath('root')\
n#2 \/var\/www\/owncloud\/lib\/private\/route\/route.php(154) : runtime-created function(1): require_once('\/var\/www\/ownclo...')\
n#3 [internal function]: __lambda_func(Array)\
n#4 \/var\/www\/owncloud\/lib\/private\/route\/router.php(273): call_user_func('?lambda_291', Array)\
n#5 \/var\/www\/owncloud\/lib\/base.php(873): OC\\Route\\Router->match('\/apps\/files_opd...')\
n#6 \/var\/www\/owncloud\/index.php(39): OC::handleRequest()\
n#7 {main}","File":"\/var\/www\/owncloud\/lib\/private\/files\/view.php","Line":1639} </code></pre>

Server Ubuntu 14.04, Apache 2.4.7, OwnCloud 9.0, php 5.5.9

**Comment by Yetangitu** [comment]
Fixed in 0.6.15, the problem was caused by a change in fileId-handling in OC9 which caused a latent bug (where fileId sometimes contained a string instead of a numerical ID) in OPDS Catalog to surface. Command strings in the ID parameter are now handled explicitly, solving this problem.

**Comment by nRaecheR** [comment]
I'm at 0.6.15 and I suffer from the same problem since my update to OC 9.0.x. Here's the error message from Moon+ :+1: 
```
No validated OPDS content, click here to see what's OPDS Catalog.


This application requires JavaScript for correct operation. Please enable JavaScript and reload the page.

ownCloud

Internal Server Error

The server encountered an internal error and was unable to complete your request.

Please contact the server administrator if this error reappears multiple times, please include the technical details below in your report.

More details can be found in the server log.


Technical details

Remote Address: 192.168.200.100Request ID: MXu6dTXr92+VpcARHnDcType: OCP\Files\NotFoundExceptionCode: 0Message: File with id "373959" has not been found.File: /var/www/owncloud/lib/private/files/view.phpLine: 1643

Trace

#0 /var/www/owncloud/lib/private/files/filesystem.php(906): OC\Files\View->getPath(373959) 
#1 /var/www/owncloud/apps/files_opds/lib/bookshelf.php(61): OC\Files\Filesystem::getPath(373959)
#2 /var/www/owncloud/apps/files_opds/lib/feed.php(63): OCA\Files_Opds\Bookshelf::get() 
#3 /var/www/owncloud/apps/files_opds/index.php(77): OCA\Files_Opds\Feed::serveFeed('E-Books', 4, 'root') 
#4 /var/www/owncloud/lib/private/route/route.php(154) : runtime-created function(1): require_once('/var/www/ownclo...') 
#5 [internal function]: __lambda_func() 
#6 /var/www/owncloud/lib/private/route/router.php(276): call_user_func('\x00lambda_463', Array) 
#7 /var/www/owncloud/lib/base.php(882): OC\Route\Router->match('/apps/files_opd...') 
#8 /var/www/owncloud/index.php(39): OC::handleRequest() 
#9 {main}

ownCloud – web services under your control
```

Are you sure the bug has been fixed with 0.6.15 or am I missing something?
